dtfs-central-api/api-tests/v1/utilisation-reports/get-selected-fee-record-details.api-test.ts

```diff
@@ -152,7 +153,7 @@ describe(`GET ${BASE_URL}`, () => {
   });
 
   function aReconciliationInProgressReport() {
-    return UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS')
+    return UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS)
```

Comment:
I'm not sure I follow what you mean, how would I even do that other than having lots of `const { RECONCILIATION_IN_PROGRESS } = UTILISATION_REPORT_RECONCILIATION_STATUS` littered all through the codebase?

Reply:
This ticket was created because we got lots of comments about not using string literals and the pattern we were asked to use instead was this I thought

Reply:
I agree with @ttbarnes .  It will be much easier to read and `RECONCILIATION_IN_PROGRESS` is very easy to understand

Reply:
That is a much larger refactor since would have to manually destructure in 156 files and would also have to so going forwards every time we use report status which is a lot.

I guess perhaps I could define the status const as follows so can simply directly import:
```
export const REPORT_NOT_RECEIVED = 'REPORT_NOT_RECEIVED';

export const PENDING_RECONCILIATION = 'PENDING_RECONCILIATION';

export const RECONCILIATION_IN_PROGRESS = 'RECONCILIATION_IN_PROGRESS';

export const RECONCILIATION_COMPLETED = 'RECONCILIATION_COMPLETED';

export const UTILISATION_REPORT_RECONCILIATION_STATUS = {
  REPORT_NOT_RECEIVED,
  PENDING_RECONCILIATION,
  RECONCILIATION_IN_PROGRESS,
  RECONCILIATION_COMPLETED,
} as const;
```
I'm not sure if that is a good pattern or not. @ChristianMcc do you have any thoughts on this?

Reply:
direct imports looks brilliant @RLavender98

Reply:
I'd be against this but open to discussion - will give my point of view below:
We're losing the context of what set of constants this is coming from within the code and we're also making it longer to navigate to what the whole set of statuses are defined as just to save a few characters on a line. For some places where this is used that isn't necessarily this case not knowing that this is a report status (especially when we might be adding statuses such as `RECORD_CORRECTION_REQUESTED`), so it's giving the reader an extra step to figure out what this constant is representing

If we're going through with this I'd agree to go with Ruby's suggestion so that the destructuring is only done once in the const file. If we're looking at destructuring within every single file it's both repetitive and it's moving context away from where this variable is used in that file

Reply:
What if we dropped the word 'RECONCILIATION' from the const? 

On the report we just have one field called 'status' so unless we are thinking we might have a second kind of status for reports in the near future we could just shorten the name a little perhaps. I wasn't on the project when it was first named so keen to know @ChristianMcc if you think that would cause any problems? 

@Zainzzkk @ttbarnes would that be an acceptable compromise?

Reply:
I think @RLavender98 's comment is perfect - keeps it simple and I don't think any context is lost:


https://github.com/UK-Export-Finance/dtfs2/pull/3784#discussion_r1822904746

But I would suggest _not_ exporting `UTILISATION_REPORT_RECONCILIATION_STATUS` so that individual statuses have to be imported and reduce LOC throughout

Reply:
> But I would suggest not exporting UTILISATION_REPORT_RECONCILIATION_STATUS so that individual statuses have to be imported and reduce LOC throughout

Surely we have to export the overall status object so we can create the type `UtilisationReportReconciliationStatus` which is really crucial to ensuring statuses are only ever set to one of the valid values?

Reply:
Current plan: do the rename suggested [here](https://github.com/UK-Export-Finance/dtfs2/pull/3784#discussion_r1823029215) and the direct exports suggested [here](https://github.com/UK-Export-Finance/dtfs2/pull/3784#discussion_r1822904746). Leave in the export of the overall constant so we can create the type and also loop over the values in parameterised tests etc which we do a fair amount. Let me know if happy with this and I will embark on the refactor.

Reply:
This change is now done. The exporting of the individual strings accidentally went on to my other branch since I had tested in there and it got merged in, didn't spot it on the PR, but have now changed to using those directly exported strings everywhere possible.

Reply:
Thanks @RLavender98 nice work